File: crates/views/src/command/environment_trash.rs
use super::user::get_user_id;
use cores::request::JsonRespnse;

#[tauri::command]
pub async fn environment_trash_query_id(
    environment_uuid: String,
) -> Result<JsonRespnse, tauri::Error> {
    let _ = get_user_id().await?;
    Ok(services_remote::requests::environment_trash::query_by_uuid(&environment_uuid).await?)
}

#[tauri::command]
pub async fn environment_trash_query(
    page_num: u32,
    page_size: u32,
) -> Result<JsonRespnse, tauri::Error> {
    let _ = get_user_id().await?;
    Ok(services_remote::requests::environment_trash::query(page_num, page_size).await?)
}

#[tauri::command]
pub async fn environment_trash_recover(
    environment_uuid: String,
) -> Result<JsonRespnse, tauri::Error> {
    let _ = get_user_id().await?;
    Ok(services_remote::requests::environment_trash::recover(&environment_uuid).await?)
}

#[tauri::command]
pub async fn environment_trash_recovers(
    environment_uuids: Vec<String>,
) -> Result<JsonRespnse, tauri::Error> {
    let _ = get_user_id().await?;
    Ok(services_remote::requests::environment_trash::recovers(environment_uuids).await?)
}

#[tauri::command]
pub async fn environment_trash_recover_all() -> Result<JsonRespnse, tauri::Error> {
    let _ = get_user_id().await?;
    Ok(services_remote::requests::environment_trash::recover_all().await?)
}

#[tauri::command]
pub async fn environment_trash_delete_batch(
    environment_uuids: Vec<String>,
) -> Result<JsonRespnse, tauri::Error> {
    let _ = get_user_id().await?;
    Ok(services_remote::requests::environment_trash::delete_batch(environment_uuids).await?)
}

#[tauri::command]
pub async fn environment_trash_clean() -> Result<JsonRespnse, tauri::Error> {
    let _ = get_user_id().await?;
    Ok(services_remote::requests::environment_trash::clean().await?)
}

File: frontend/src/commands/environment-trash.ts
import { invoke } from '@tauri-apps/api/core'

// 根据环境UUID查询回收站中的环境
export const environment_trash_query_id = async (environmentUuid: String): Promise<any> => {
    return await invoke('environment_trash_query_id', { environmentUuid })
};

// 分页查询回收站中的环境列表
export const environment_trash_query = async (pageNum: number, pageSize: number): Promise<any> => {
    return await invoke('environment_trash_query', { pageNum, pageSize })
};

// 恢复单个环境
export const environment_trash_recover = async (environmentUuid: string): Promise<any> => {
    return await invoke('environment_trash_recover', { environmentUuid })
};

// 批量恢复多个环境
export const environment_trash_recovers = async (environmentUuids: String[]): Promise<any> => {
    return await invoke('environment_trash_recovers', { environmentUuids })
};

// 恢复回收站中的所有环境
export const environment_trash_recover_all = async (): Promise<any> => {
    return await invoke('environment_trash_recover_all', {})
};

// 批量删除回收站中的环境
export const environment_trash_delete_batch = async (environmentUuids: String[]): Promise<any> => {
    return await invoke('environment_trash_delete_batch', { environmentUuids })
};

// 清空回收站
export const environment_trash_clean = async (
): Promise<any> => {
    return await invoke('environment_trash_clean', {})
};